src/components/MediaSettings/VideoBackgroundEditor.vue

```diff
@@ -110,10 +140,111 @@ export default {
+
+		// Create the backgrounds folder if it doesn't exist
+		if (await client.exists(absoluteBackgroundsFolderPath) === false) {
+			await client.createDirectory(absoluteBackgroundsFolderPath)
+		}
+
+		// Create picker
+		picker = getFilePickerBuilder(t('spreed', 'File to share'))
+			.setMultiSelect(false)
+			.setModal(true)
+			.startAt(relativeBackgroundsFolderPath)
+			.setType(1)
+			.allowDirectories(false)
+			.build()
+
+		this.loadBackground()
+	},
+
 	methods: {
-		handleSelectBackground(background) {
-			this.$emit('update-background', background)
-			this.selectedBackground = background
+		handleSelectBackground(path) {
+			this.$emit('update-background', path)
+			this.selectedBackground = path
+		},
+
+		/**
+		 * Clicks the hidden file input and opens the file-picker
+		 */
+		clickImportInput() {
+			this.$refs.fileUploadInput.click()
+		},
+
+		async handleFileInput(event) {
+
+			// Make file path
+			const file = Object.values(event.target.files)[0]
+
+			// Clear input to ensure that the change event will be emitted if
+			// the same file is picked again.
+			event.target.value = ''
+
+			const fileName = file.name
+
+			// userRoot path
+			const userRoot = '/files/' + this.$store.getters.getUserId()
+
+			const filePath = this.$store.getters.getAttachmentFolder() + '/Backgrounds/' + fileName
+
+			// Get a unique relative path based on the previous path variable
+			const uniquePath = await findUniquePath(client, userRoot, filePath)
+
+			try {
+				// Upload the file
+				await client.putFileContents(userRoot + uniquePath, file, {
+					contentLength: file.size,
+				})
+
+				const previewURL = await generateUrl('/core/preview.png?file={path}&x=-1&y={height}&a=1', {
+					path: filePath,
+					height: 1080,
+				})
+
+				this.handleSelectBackground(previewURL)
+
+			} catch (error) {
+				console.debug(error)
+				showError(t('spreed', 'Error while uploading the file'))
+			}
+		},
+
+		openPicker() {
+			picker.pick()
+				.then((path) => {
+					if (!path.startsWith('/')) {
+						throw new Error(t('files', 'Invalid path selected'))
+					}
+
+					const previewURL = generateUrl('/core/preview.png?file={path}&x=-1&y={height}&a=1', {
+						path,
+						height: 1080,
+					})
+
+					this.handleSelectBackground(previewURL)
+				})
+		},
+
+		loadBackground() {
+			// Set virtual background depending on localstorage's settings
+			if (localStorage.getItem('virtualBackgroundEnabled_' + this.token) === 'true') {
+				if (localStorage.getItem('virtualBackgroundType_' + this.token) === VIRTUAL_BACKGROUND.BACKGROUND_TYPE.BLUR) {
+					this.selectedBackground = 'blur'
+				} else if (localStorage.getItem('virtualBackgroundType_' + this.token) === VIRTUAL_BACKGROUND.BACKGROUND_TYPE.IMAGE) {
+					this.selectedBackground = localStorage.getItem('virtualBackgroundUrl_' + this.token)
+				}
+			} else {
+				this.selectedBackground = 'none'
+			}
 		},
```

Comment:
**question:** @nextcloud/browser-storage

I may be wrong, but it seems, we always use `@nextcloud/browser-storage` with `services/BrowserStorage.js`

Reply:
**question:** constants as values

May we also use `VIRTUAL_BACKGROUND.BACKGROUND_TYPE` as values here instead of raw strings `'blur'`, `'none'`